### PR TITLE
Daemon: passdown the `--oom-kill-disable` option to containerd

### PR DESCRIPTION
**Pull request by arm64b**
Current implementaion of docke daemon doesn't pass down the
`--oom-kill-disable` option specified by the end user to the containerd
when spawning a new docker instance with help from `runc` component, which
results in the `--oom-kill-disable` doesn't work no matter the flag is `true`
or `false`.

This PR will fix this issue reported by #36090

Signed-off-by: Dennis Chen <dennis.chen@arm.com>
Signed-off-by: Jianyong Wu <jianyong.wu@arm.com>

**- What I did**
Fix `--oom-kill-disable` issue
**- How I did it**
Pass down the actual value of the `--oom-kill-disable` specified by the end user to the containerd
**- How to verify it**
Run a container which will use up all the limited memory soon, like this:
`/home/dennis/test# docker run -d -v "$PWD:/tmp" --oom-kill-disable=true -m 10MB  --name oomk-disable busybox sh -c /tmp/oom.sh`

`oom.sh` is a script to use up the memory quickly:
```
#!/bin/sh
x=a
while true
do
        x=$x$x$x$x
        echo "hello world"
        sleep 4
done   
```
We can verify this in 2 methods:
1. When the new docker instance spawned, there is a new generated folder under the `/sys/fs/cgroup/memory/docker` looks like `/sys/fs/cgroup/memory/docker/423797dba327678cd5c3e4eb5e8723019c1aaeface3fab3755be3fd454b36076` in my system, then we can `cat memory.oom_control` file in this folder to verify if the `--oom-kill-disable` works or not. 
Before this PR, the `oom_kill_disable` filed in the file `memory.oom_control` is always 0 no matter we `docker run` a container with `--oom-kill-disable` as `true` or `false`. After this PR applied, the `memory.oom_control` will be changed according to the actual value assigned by the end user.

2. Launch another terminal then run `docker events` to watch the events when running `/home/dennis/test# docker run -d -v "$PWD:/tmp" --oom-kill-disable=true -m 10MB  --name oomk-disable busybox sh -c /tmp/oom.sh`
Without this PR, the events will be:
```
...
container start
container oom
container die
...
```
With this PR, the events will be:
```
container start
container oom
container oom
container oom
...
```
We can see that the `--oom-kill-disable=true` works after applied this PR, consequently the `DockerSuite.TestEventsOOMDisableFalse` test case passed.

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**
summit

### CODE REVIEW

**Review by dnephin — APPROVED** [review]
LGTM

**Comment by tophj-ibm** [comment]
Just tested, fix works for me.

**Comment by arm64b** [comment]
Thanks @dnephin for the review 👍 
Let's ping the maintainers who commit this file recently for review to get another approval . @cpuguy83 @thaJeztah

**Comment by thaJeztah** [comment]
Looks like this was introduced in https://github.com/moby/moby/commit/45d85c99139bbd16004bbedb7d5bac6a60264538 (https://github.com/moby/moby/pull/34356), which is part of 17.09

ping @mlaventure 

Should we have a unit-test for this?

**Comment by arm64b** [comment]
Hello @thaJeztah , seems we've already have a corresponding test file for the `daemon_unix.go`, this RP just passes a `oom-` flag into containerd, we certainly have a  `DockerSuite.TestEventsOOMDisableFalse` in 
integration which can have effective test for the change, so IMO we can postpone this unit now. But I flexible for this if we do need.

**Comment by thaJeztah** [comment]
> we certainly have a DockerSuite.TestEventsOOMDisableFalse in
integration which can have effective test for the change

Yes, so I was wondering why none of the tests caught this

**Comment by arm64b** [comment]
Ah, the reason I have mentioned in PR #36185 ` Interesting all the arches (amd64, s390, ppc...) have skipped this test, I don't know the context/story here... 🤕` Let's abstract the test result of the `TestEventsOOMDisableTrue` on log of `Janky`:
```
18:28:26 SKIP: docker_cli_events_unix_test.go:51: DockerSuite.TestEventsOOMDisableFalse (unmatched requirement swapMemorySupport)
18:28:26 SKIP: docker_cli_events_unix_test.go:81: DockerSuite.TestEventsOOMDisableTrue (unmatched requirement swapMemorySupport)
```
That comes from the `testRequires` of this test case:
```
func (s *DockerSuite) TestEventsOOMDisableTrue(c *check.C) {
	testRequires(c, DaemonIsLinux, oomControl, memoryLimitSupport, NotArm, swapMemorySupport) 
```
I think we can remove the `swapMemorySupport` and see what happens after this PR merged.

**Comment by arm64b** [comment]
BTW: on our Linux/arm64 system, `swapMemorySupport` is true, so that triggers the issue.

**Review by thaJeztah — APPROVED** [review]
Ah right; that would explain

LGTM

**Comment by arm64b** [comment]
Thanks @thaJeztah for the quick response 👍

**Comment by thaJeztah** [comment]
Yw! Can you do a follow up to update the test conditions (as discussed above?)

**Comment by arm64b** [comment]
> Yw! Can you do a follow up to update the test conditions (as discussed above?)

Absolutely. Please see PR #36229 😄

**Pull request merged**
